app/views/criteria/_new_criterion.html.erb

```diff
@@ -1,8 +1,8 @@
 <li id='new_criterion'>
 	<%= form_for @criterion,
-    url: { controller: "#{criterion_type.to_s.tableize}",
-           action: 'index',
-           assignment_id: assignment.id },
+    url: assignment_criteria_path(assignment_id: assignment.id,
+                                  criterion_type: @criterion_type),
```

Comment:
In general you should try to avoid using `@criterion_type` if you already have access to the criterion object.